### DISCUSSION TITLE
principles/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#54] [Announcements]
# principles/

Programming Principles There are certain universal laws and principles in software development that guide architects, programmers, and anyone needing to design software. This page lists quite a few of those principles, although it's far from complete. This page is a fork of programming-principles repository by Lars Kappert, who has done most of the work collecting the material.

https://java-design-patterns.com/principles/

**Comment by saman488** [comment]
I was looking for Design Patterns in java. Thank you I am still learning.

**Comment by rajshreeee** [comment]
Awesome!

**Comment by RSCloud2233** [comment]
It's very useful for me. Thank you for sharing!

**Comment by Zelechos** [comment]
Elegante, consiso y simple realmente es un contenido de muy buena calidad y breve

**Comment by RSCloud2233** [comment]


**Reply by vuhung3990** [reply]
đẳng cấp

**Comment by somixyz** [comment]
Really useful material

**Comment by SEKSEF** [comment]
Thank you for this masterpiece!
